components/tournaments_card/commons/tournaments_card_base.lua

```diff
@@ -0,0 +1,659 @@
+
+function BaseTournamentsCard:build()
+	if not self.data then
+		return
+	end
+
+	self.display = mw.html.create('div')
+		:addClass('gridTable tournamentCard')
+
+	local config = self.config
+
+	if not config.showTier then
+		self.display:addClass('Tierless')
+	end
+
+	if config.showRank then
+		self.display:addClass('Ranked')
+	end
+
+	if config.showQualifierColumnOverWinnerRunnerup then
+		self.display:addClass('Qualifiers')
+	end
+
+	if not config.showGameIcon then
+		self.display:addClass('NoGameIcon')
+	end
+
+	if config.showOrganizer then
+		self.display:addClass('HasOrganizer')
+	end
+
+	self.display:node(self:_header())
+
+	self.cachedData = {rank = 1, prize = 0, skippedRanks = self.config.offset}
+	for _, rowData in ipairs(self.data) do
+		self:_row(rowData)
+	end
+
+	return self.display
+end
+
+function BaseTournamentsCard:_header()
+	local config = self.config
+
+	local header = mw.html.create('div'):addClass('gridHeader')
+
+	if config.showRank then
+		header:tag('div'):addClass('gridCell Position'):wikitext('#')
+	end
+
+	if config.showTier then
+		header:tag('div'):addClass('gridCell Tier'):wikitext('Tier')
+	end
+
+	local gameHeader = header:tag('div'):addClass('gridCell')
+	if config.showGameIcon then
+		gameHeader:addClass('GameSeries'):wikitext(Abbreviation.make('G & S', 'Game and Series'))
+	else
+		gameHeader:addClass('Series'):wikitext(Abbreviation.make('S', 'Series'))
+	end
+
+	header:tag('div'):addClass('gridCell'):wikitext('Tournament'):done()
+
+	if config.showOrganizer then
+		header:tag('div'):addClass('gridCell'):wikitext('Organizer')
+	end
+
+	header
+		:tag('div'):addClass('gridCell'):wikitext('Date'):done()
+		:tag('div'):addClass('gridCell'):wikitext('Prize' .. NONBREAKING_SPACE .. 'Pool'):done()
+		:tag('div'):addClass('gridCell'):wikitext('Location'):done()
+		:tag('div'):addClass('gridCell'):wikitext(Abbreviation.make('P#', 'Number of Participants'))
+
+	if config.showQualifierColumnOverWinnerRunnerup then
+		header:tag('div'):addClass('gridCell'):wikitext('Qualified')
+	else
+		header
+			:tag('div'):addClass('gridCell'):wikitext('Winner'):done()
+			:tag('div'):addClass('gridCell'):wikitext('Runner-up')
+	end
+
+	return header
+end
+
+function BaseTournamentsCard:_row(tournamentData)
+	local config = self.config
+
+	local highlight = config.showHighlight and self:getHighlightClass(tournamentData) or nil
+	local status = tournamentData.status and tournamentData.status:lower()
+
+	local row = mw.html.create('div')
+		:addClass('gridRow')
+		:addClass(highlight)
+
+	if config.showRank then
+		self:_calculateRank(tonumber(tournamentData.prizepool) or 0)
+
+		row:tag('div'):addClass('gridCell Position Header')
+			:node(self.cachedData.rank)
+	end
+
+	local gameIcon = config.showGameIcon
+		and mw.html.create('span'):addClass('icon-16px GameIcon'):wikitext(Game.icon{
+			game = tournamentData.game, useDefault = false
+		}) or nil
+
+	if config.showTier then
+		row:tag('div')
+			:addClass('gridCell Tier Header')
```

Comment:
Why Header class for a row item?

Reply:
this has 2 displays
1 for on desktop where it is displayed as a row
and one for on mmobile where it is displayed as cards
the headers in a "row" are headers in the card display

![Screenshot 2023-04-24 16 33 44](https://user-images.githubusercontent.com/75081997/234029013-437bd53e-aec2-4b2b-89a9-8860a6f141fb.png)
![Screenshot 2023-04-24 16 33 57](https://user-images.githubusercontent.com/75081997/234029016-82440152-ee23-4511-ab95-4c102d875932.png)